packages/ui/src/utils/typeUtils.tsx

```diff
@@ -0,0 +1,15 @@
+/**
+ * Valid JavaScript type names that can be returned by the typeof operator
+ */
+type JavaScriptType = 'string' | 'number' | 'boolean' | 'undefined' | 'object' | 'function' | 'symbol' | 'bigint'
+
+/**
+ * Checks if a value is of one of the provided types
+ * @param value - The value to check
+ * @param types - Array of JavaScript type names to check against
+ * @returns True if the value matches any of the provided types, false otherwise
+ */
+export const anyTypeOf = (value: unknown, types: JavaScriptType[]): boolean => {
```

Comment:
I understand the idea of this utility function, but unfortunately TS won't follow the logic though and assert the correct type on the other side (in the version of TS we're on). It's probably safer to use `typeof myThing === 'string' || typeof myThing === 'number'` when the types matter. I wonder if it would be possible to write something generic like this that asserts the types using the `value is string | number` return syntax 🤔

Reply:
I asked AI to do what you wondered for the TS version we use, and this is what he came up with. Looks nice

<img width="812" alt="image" src="https://github.com/user-attachments/assets/9f825e78-7857-45c0-b412-8055762b154a" />